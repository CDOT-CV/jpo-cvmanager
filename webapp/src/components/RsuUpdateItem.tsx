import React, { Component } from 'react'
import './css/RsuUpdateItem.css'

export type RsuUpdateItemProps = {
  ip: string
  osUpdateAvailable: string[]
  fwUpdateAvailable: string[]
  handleUpdateOS: (ip: string) => void
  handleUpdateFW: (ip: string) => void
}

const RsuUpdateItem = (props: RsuUpdateItemProps) => {
  const handleOSClick = () => {
    props.handleUpdateOS(props.ip)
  }

  const handleFWClick = () => {
    props.handleUpdateFW(props.ip)
  }

  return (
    <div id="updateitem">
      <div id="item">
        {<h3 id="ipaddr">{props.ip}</h3>}

        {props.osUpdateAvailable.includes(props.ip) ? (
          <button id="updatebtn" onClick={handleOSClick}>
            Update OS
          </button>
        ) : (
          <button id="disabledbtn" disabled={true}>
            Update OS
          </button>
        )}

        {!props.osUpdateAvailable.includes(props.ip) && props.fwUpdateAvailable.includes(props.ip) ? (
          <button id="updatebtn" onClick={handleFWClick}>
            Update Firmware
          </button>
        ) : (
          <button id="disabledbtn" disabled={true}>
            Update Firmware
          </button>
        )}

        {props.osUpdateAvailable.includes(props.ip) || props.fwUpdateAvailable.includes(props.ip) ? (
          <p id="infotext">Allow for 10 minutes for update to occur once started</p>
        ) : null}
      </div>
    </div>
  )
}

export default RsuUpdateItem
