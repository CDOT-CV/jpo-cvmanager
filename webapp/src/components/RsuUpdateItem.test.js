import React from "react";
import { render } from "@testing-library/react";
import RsuUpdateItem from "./RsuUpdateItem";

it("should take a snapshot", () => {
  const { asFragment } = render(<RsuUpdateItem osUpdateAvailable={[]} fwUpdateAvailable={[]} />);

  expect(asFragment(<RsuUpdateItem osUpdateAvailable={[]} fwUpdateAvailable={[]} />)).toMatchSnapshot();
});
