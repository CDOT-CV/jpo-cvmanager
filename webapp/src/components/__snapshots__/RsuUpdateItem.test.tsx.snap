// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot handleUpdateFW 1`] = `
<div>
  <div
    id="updateitem"
  >
    <div
      id="item"
    >
      <h3
        id="ipaddr"
      >
        1.1.1.1
      </h3>
      <button
        disabled=""
        id="disabledbtn"
      >
        Update OS
      </button>
      <button
        id="updatebtn"
      >
        Update Firmware
      </button>
      <p
        id="infotext"
      >
        Allow for 10 minutes for update to occur once started
      </p>
    </div>
  </div>
</div>
`;

exports[`snapshot osUpdateAvailable 1`] = `
<div>
  <div
    id="updateitem"
  >
    <div
      id="item"
    >
      <h3
        id="ipaddr"
      >
        1.1.1.1
      </h3>
      <button
        id="updatebtn"
      >
        Update OS
      </button>
      <button
        disabled=""
        id="disabledbtn"
      >
        Update Firmware
      </button>
      <p
        id="infotext"
      >
        Allow for 10 minutes for update to occur once started
      </p>
    </div>
  </div>
</div>
`;
